plugins/outputs/inlong/inlong.go

```diff
@@ -0,0 +1,88 @@
+package inlong
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/apache/inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const managerURLSuffix = "/inlong/manager/openapi/dataproxy/getIpList"
+
+type Inlong struct {
+	GroupID    string          `toml:"group_id"`
+	StreamID   string          `toml:"stream_id"`
+	ManagerURL string          `toml:"manager_url"`
+	Log        telegraf.Logger `toml:"-"`
+
+	producerFunc func(groupId string, managerUrl string) (dataproxy.Client, error)
```

Comment:
This indirection through `producerFunc` seems to be introduced for testing. However, as it stands now, the tests add no real value so I would remove the indirection in favor of simplifying the code.

Reply:
Well, I will use it at TestInlong_Connect